### ISSUE TITLE
Cannot write to DB after migration from version 1.4.4 to 1.5.0

### ISSUE THREAD

**Issue by mariol3** [neo4j/neo4j-java-driver#458]
Hi,
We migrated from version 1.4.4 to 1.5.0 on existing code to solve a problem related to pending db connections but we found that our previous working code to write some nodes to DB now is not writing them anymore, without giving any errors.

This is the existing code that was working with version 1.4.4
```
        ...
        Transaction tx = null;
        try (Session session = driver.session()) {
            tx = session.beginTransaction();
            createdNode = tx.run('SOME_WRITE_QUERY');
            tx.success();
        } catch (Exception e) {
            if (tx != null) tx.failure();
            log.error(this.getClass().getName(), e);
            throw new RuntimeException(e);
        } finally {
            if ((tx != null) && (tx.isOpen())) {
                tx.close();
            }
       }
       ...
```

While trying to solve the problem we founded that by adding a call to `tx.commitAsync();` after the call to `tx.success();` it worked.

Are we doing something wrong? There is something new with the 1.5.0 driver that we don't know?
Thanks for your response!

**Comment by lutovich** [comment]
Hi @mariol3,

I think this is an unintentional change in behaviour. Your code allows `Session` to be closed before `Transaction`, which is an unusual pattern. Driver 1.5.0 will simply rollback any uncommitted transaction during `Session#close()`.

You could change your code to something like this:

```java
try ( Session session = driver.session();
      Transaction tx = session.beginTransaction() )
{
    createdNode = tx.run( 'SOME_WRITE_QUERY' );
    tx.success();
}
catch ( Exception e )
{
    log.error( this.getClass().getName(), e );
    throw new RuntimeException( e );
}
```

where transaction is closed and then session is closed. This code is a bit shorter and more close to the recommended/expected usage pattern. Do you think it is an option for you?

We will think more about this behaviour and if it should be fixed or changed.

**Comment by mariol3** [comment]
Hi @lutovich ,

I tried your solution and it works with my current code 💪 
Thank you for your explanation of the problem and support!

**Comment by lutovich** [comment]
Hi @mariol3,

Glad it works for you!
Previous behaviour should be restored by the attached PR. However I'd still recommend using `Session` and `Transaction` in a single try-with-resources block because code feels better that way.

**Comment by mariol3** [comment]
Cool @lutovich thank you for the fix! 
Yeah I agree with you that code feels better with your try-with-resources block, I'll keep it that way.

**Comment by lutovich** [comment]
Mentioned PR is now available with 1.5.1 release of the driver.
